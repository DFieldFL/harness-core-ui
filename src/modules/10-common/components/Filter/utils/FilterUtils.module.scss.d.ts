/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly noStyleUl: string
  readonly summaryItem: string
}
export default styles
