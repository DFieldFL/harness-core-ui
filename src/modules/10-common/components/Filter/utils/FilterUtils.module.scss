.noStyleUl {
  list-style-type: none;
  padding: var(--spacing-small);
  margin: 0;
  background-color: var(--black);
  color: var(--white);
}

.summaryItem {
  padding: var(--spacing-xsmall);
  font-size: small;
}
