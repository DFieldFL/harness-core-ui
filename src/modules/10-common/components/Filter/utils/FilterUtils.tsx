import { isEmpty, isUndefined, omitBy } from 'lodash-es'
import React from 'react'
import type { FilterDTO } from 'services/cd-ng'

import css from './FilterUtils.module.scss'

type supportedTypes = string | number | boolean | object

const tagSeparator = ' : '

export const renderItemByType = (data: supportedTypes | Array<supportedTypes> | object): string => {
  if (Array.isArray(data)) {
    return data.join(', ')
  } else if (typeof data === 'object') {
    return Object.entries(data as object)
      .map(([key, value]) => {
        return key.toString().concat(value ? tagSeparator.concat(value.toString()) : '')
      })
      .join(', ')
  } else if (typeof data === 'number') {
    return data.toString()
  } else if (typeof data === 'boolean') {
    return data ? 'true' : 'false'
  }
  return typeof data === 'string' ? data : ''
}

export const getFilterSummary = (fieldToLabelMapping: Map<string, string>, fields: object): JSX.Element => {
  return (
    <ol className={css.noStyleUl}>
      {Object.entries(fields as object).map(([key, value]) => {
        if (fieldToLabelMapping.has(key)) {
          return (
            <li key={key} className={css.summaryItem}>
              <span>{fieldToLabelMapping.get(key)} : </span>
              <span style={{ fontWeight: 'bold' }}>{renderItemByType(value)}</span>
            </li>
          )
        }
      })}
    </ol>
  )
}

export const removeNullAndEmpty = (object: Record<string, any>) => omitBy(omitBy(object, isUndefined), isEmpty)
export const isObjectEmpty = (arg: Record<string, any>): boolean => isEmpty(omitBy(arg, isEmpty))
export const UNSAVED_FILTER = 'Unsaved Filter'
export const MAX_FILTER_NAME_LENGTH = 25

export const flattenObject = (obj: Record<string, any>) => {
  const flattened: {
    [key: string]: string
  } = {}
  Object.keys(obj).forEach((key: string) => {
    if (Object.prototype.toString.call(obj[key]) === '[object Object]' && obj[key] !== null) {
      Object.assign(flattened, flattenObject(obj[key]))
    } else {
      flattened[key as string] = obj[key]
    }
  })
  return flattened
}

export const getFilterSize = (obj: Record<string, any>): number => {
  return Object.keys(flattenObject(obj)).length
}

export const getFilterByIdentifier = (filters: FilterDTO[], identifier: string): FilterDTO | undefined =>
  filters?.find((filter: FilterDTO) => filter.identifier?.toLowerCase() === identifier.toLowerCase())
