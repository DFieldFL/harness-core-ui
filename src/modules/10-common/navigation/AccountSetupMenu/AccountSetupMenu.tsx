import React from 'react'
import { Layout } from '@wings-software/uicore'
import { useParams } from 'react-router-dom'
import routes from '@common/RouteDefinitions'
import { useFeatureFlags } from '@common/hooks/useFeatureFlag'
import type { AccountPathProps } from '@common/interfaces/RouteInterfaces'
import { useStrings } from 'framework/strings'
import { SidebarLink } from '../SideNav/SideNav'
import NavExpandable from '../NavExpandable/NavExpandable'

const AccountSetupMenu: React.FC = () => {
  const { getString } = useStrings()
  const { accountId } = useParams<AccountPathProps>()
  const { NG_RBAC_ENABLED, NG_SIGNUP } = useFeatureFlags()
  return (
    <NavExpandable title={getString('common.accountSetup')} route={routes.toSetup({ accountId })}>
      <Layout.Vertical spacing="small">
        <SidebarLink exact label={getString('overview')} to={routes.toSetup({ accountId })} />
        <SidebarLink label={getString('authentication')} to={routes.toAuthenticationSettings({ accountId })} />
        <SidebarLink label={getString('connectorsLabel')} to={routes.toConnectors({ accountId })} />
        <SidebarLink label={getString('common.secrets')} to={routes.toSecrets({ accountId })} />
        {NG_RBAC_ENABLED ? (
          <SidebarLink to={routes.toAccessControl({ accountId })} label={getString('accessControl')} />
        ) : null}
        {NG_SIGNUP && (
          <SidebarLink exact label={getString('common.subscriptions')} to={routes.toSubscriptions({ accountId })} />
        )}
        <SidebarLink label={getString('orgsText')} to={routes.toOrganizations({ accountId })} />
      </Layout.Vertical>
    </NavExpandable>
  )
}

export default AccountSetupMenu
