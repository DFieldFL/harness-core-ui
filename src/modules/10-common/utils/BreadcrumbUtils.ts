import type { Breadcrumb } from '@wings-software/uicore'
import type { UseStringsReturn } from 'framework/strings'
import { getScopeFromDTO } from '@common/components/EntityReference/EntityReference'
import { Scope } from '@common/interfaces/SecretsInterface'
import routes from '@common/RouteDefinitions'

interface GetLinkForAccountResourcesProps {
  accountId: string
  orgIdentifier?: string
  projectIdentifier?: string
  getString: UseStringsReturn['getString']
}

export const getLinkForAccountResources = ({
  accountId,
  orgIdentifier,
  projectIdentifier,
  getString
}: GetLinkForAccountResourcesProps): Breadcrumb[] => {
  return getScopeFromDTO({ orgIdentifier, projectIdentifier }) === Scope.ACCOUNT
    ? [{ label: getString('common.accountResources'), url: routes.toAccountResources({ accountId }) }]
    : []
}
