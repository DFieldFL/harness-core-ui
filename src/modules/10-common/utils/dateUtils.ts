export const formatDatetoLocale = (date: number | string): string => {
  return `${new Date(date).toLocaleDateString()} ${new Date(date).toLocaleTimeString()}`
}
