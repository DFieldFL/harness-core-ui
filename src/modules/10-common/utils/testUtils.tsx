import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation } from 'react-router-dom'
import { ModalProvider } from '@wings-software/uikit'
import qs from 'qs'

import strings from 'strings/strings.en.yaml'
import { AppStoreProvider, useAppStoreWriter } from 'framework/hooks/useAppStore'
import type { AppStore } from 'framework/types/AppStore'
import { AUTH_ROUTE_PATH_PREFIX, StringsContext } from 'framework/exports'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): Element | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): Element | null => document.querySelector('.bp3-popover-content')

interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, any>
  defaultAppStoreValues?: Partial<AppStore>
}

export const prependAccountPath = (path: string): string => AUTH_ROUTE_PATH_PREFIX + path

const AppStoreInitializer: React.FC<{ defaultAppStoreValues: TestWrapperProps['defaultAppStoreValues'] }> = ({
  children,
  defaultAppStoreValues
}) => {
  const updateAppStore = useAppStoreWriter()
  updateAppStore(defaultAppStoreValues || {})
  return <> {children} </>
}

export const NotFound = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  const { path = '/', pathParams = {}, defaultAppStoreValues, queryParams = {} } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  const history = createMemoryHistory({ initialEntries: [routePath] })

  return (
    <AppStoreProvider>
      <StringsContext.Provider value={strings}>
        <AppStoreInitializer defaultAppStoreValues={defaultAppStoreValues}>
          <Router history={history}>
            <ModalProvider>
              <RestfulProvider base="/">
                <Switch>
                  <Route exact path={path}>
                    {props.children}
                  </Route>
                  <Route>
                    <NotFound />
                  </Route>
                </Switch>
              </RestfulProvider>
            </ModalProvider>
          </Router>
        </AppStoreInitializer>
      </StringsContext.Provider>
    </AppStoreProvider>
  )
}
