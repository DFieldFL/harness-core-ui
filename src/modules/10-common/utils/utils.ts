import type { IconName } from '@wings-software/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}
