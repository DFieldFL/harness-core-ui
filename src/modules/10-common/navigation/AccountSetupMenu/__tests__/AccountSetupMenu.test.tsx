import React from 'react'
import { fireEvent, render, RenderResult, waitFor } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import AccountSetupMenu from '../AccountSetupMenu'

const currentUser = {
  accounts: [
    {
      uuid: '123',
      createdFromNG: true
    }
  ]
}

describe('Account Setup Menu', () => {
  let container: HTMLElement
  let getByText: RenderResult['getByText']

  beforeEach(async () => {
    const renderObj = render(
      <TestWrapper
        path="/account/:accountId"
        pathParams={{ accountId: '123' }}
        defaultAppStoreValues={{ currentUserInfo: currentUser }}
      >
        <AccountSetupMenu />
      </TestWrapper>
    )
    container = renderObj.container
    getByText = renderObj.getByText
  })
  test('it renders the chevron', () => {
    expect(container).toMatchSnapshot()
  })
  test('it renders the dropdown with the overview link', async () => {
    const accountSetup = getByText('common.accountSetup')
    fireEvent.mouseEnter(accountSetup)

    await waitFor(() => {
      expect(getByText('overview')).toBeTruthy
    })

    expect(container).toMatchSnapshot()
  })
  test('it renders the dropdown with the authentication link', async () => {
    const accountSetup = getByText('common.accountSetup')
    fireEvent.mouseEnter(accountSetup)

    await waitFor(() => {
      expect(getByText('authentication')).toBeTruthy
    })

    expect(container).toMatchSnapshot()
  })
  test('it renders the dropdown with the connectors link', async () => {
    const accountSetup = getByText('common.accountSetup')
    fireEvent.mouseEnter(accountSetup)

    await waitFor(() => {
      expect(getByText('connectorsLabel')).toBeTruthy
    })

    expect(container).toMatchSnapshot()
  })
  test('it renders the dropdown with the secrets link', async () => {
    const accountSetup = getByText('common.accountSetup')
    fireEvent.mouseEnter(accountSetup)

    await waitFor(() => {
      expect(getByText('common.secrets')).toBeTruthy
    })

    expect(container).toMatchSnapshot()
  })
  test('it renders the dropdown with the access control link', async () => {
    const accountSetup = getByText('common.accountSetup')
    fireEvent.mouseEnter(accountSetup)

    await waitFor(() => {
      expect(getByText('accessControl')).toBeTruthy
    })

    expect(container).toMatchSnapshot()
  })
  test('it renders the dropdown with the subscriptions link', async () => {
    const accountSetup = getByText('common.accountSetup')
    fireEvent.mouseEnter(accountSetup)

    await waitFor(() => {
      expect(getByText('common.subscriptions.title')).toBeTruthy
    })

    expect(container).toMatchSnapshot()
  })
  test('it renders the dropdown with the organizations link', async () => {
    const accountSetup = getByText('common.accountSetup')
    fireEvent.mouseEnter(accountSetup)

    await waitFor(() => {
      expect(getByText('orgsText')).toBeTruthy
    })

    expect(container).toMatchSnapshot()
  })
})
