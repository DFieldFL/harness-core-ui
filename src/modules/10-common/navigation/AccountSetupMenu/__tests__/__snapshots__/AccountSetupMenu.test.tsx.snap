// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Setup Menu it renders the chevron 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main main"
  >
    <p
      class="StyledProps--font StyledProps--main text StyledProps--font-xsmall StyledProps--flex StyledProps--flex-alignItems-flex-start StyledProps--flex-justifyContent-space-between"
      style="display: flex; align-items: center;"
    >
      common.accountSetup
      <span
        class="bp3-icon bp3-icon-chevron-up StyledProps--main StyledProps--padding-left-xsmall"
        icon="chevron-up"
      >
        <svg
          data-icon="chevron-up"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            chevron-up
          </desc>
          <path
            d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 001.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </p>
  </div>
</div>
`;

exports[`Account Setup Menu it renders the dropdown with the access control link 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main main"
  >
    <p
      class="StyledProps--font StyledProps--main text StyledProps--font-xsmall StyledProps--flex StyledProps--flex-alignItems-flex-start StyledProps--flex-justifyContent-space-between"
      style="display: flex; align-items: center;"
    >
      common.accountSetup
      <span
        class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-left-xsmall"
        icon="chevron-down"
      >
        <svg
          data-icon="chevron-down"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            chevron-down
          </desc>
          <path
            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </p>
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
    >
      <a
        class="link"
        href="/account/123/home/setup/"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          overview
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/authentication"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          authentication
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/connectors"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          connectorsLabel
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/secrets"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.secrets
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/access-control"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          accessControl
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/subscriptions"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.subscriptions.title
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/organizations"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          orgsText
        </p>
      </a>
    </div>
  </div>
</div>
`;

exports[`Account Setup Menu it renders the dropdown with the authentication link 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main main"
  >
    <p
      class="StyledProps--font StyledProps--main text StyledProps--font-xsmall StyledProps--flex StyledProps--flex-alignItems-flex-start StyledProps--flex-justifyContent-space-between"
      style="display: flex; align-items: center;"
    >
      common.accountSetup
      <span
        class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-left-xsmall"
        icon="chevron-down"
      >
        <svg
          data-icon="chevron-down"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            chevron-down
          </desc>
          <path
            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </p>
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
    >
      <a
        class="link"
        href="/account/123/home/setup/"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          overview
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/authentication"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          authentication
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/connectors"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          connectorsLabel
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/secrets"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.secrets
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/access-control"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          accessControl
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/subscriptions"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.subscriptions.title
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/organizations"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          orgsText
        </p>
      </a>
    </div>
  </div>
</div>
`;

exports[`Account Setup Menu it renders the dropdown with the connectors link 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main main"
  >
    <p
      class="StyledProps--font StyledProps--main text StyledProps--font-xsmall StyledProps--flex StyledProps--flex-alignItems-flex-start StyledProps--flex-justifyContent-space-between"
      style="display: flex; align-items: center;"
    >
      common.accountSetup
      <span
        class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-left-xsmall"
        icon="chevron-down"
      >
        <svg
          data-icon="chevron-down"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            chevron-down
          </desc>
          <path
            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </p>
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
    >
      <a
        class="link"
        href="/account/123/home/setup/"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          overview
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/authentication"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          authentication
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/connectors"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          connectorsLabel
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/secrets"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.secrets
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/access-control"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          accessControl
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/subscriptions"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.subscriptions.title
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/organizations"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          orgsText
        </p>
      </a>
    </div>
  </div>
</div>
`;

exports[`Account Setup Menu it renders the dropdown with the organizations link 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main main"
  >
    <p
      class="StyledProps--font StyledProps--main text StyledProps--font-xsmall StyledProps--flex StyledProps--flex-alignItems-flex-start StyledProps--flex-justifyContent-space-between"
      style="display: flex; align-items: center;"
    >
      common.accountSetup
      <span
        class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-left-xsmall"
        icon="chevron-down"
      >
        <svg
          data-icon="chevron-down"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            chevron-down
          </desc>
          <path
            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </p>
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
    >
      <a
        class="link"
        href="/account/123/home/setup/"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          overview
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/authentication"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          authentication
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/connectors"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          connectorsLabel
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/secrets"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.secrets
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/access-control"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          accessControl
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/subscriptions"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.subscriptions.title
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/organizations"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          orgsText
        </p>
      </a>
    </div>
  </div>
</div>
`;

exports[`Account Setup Menu it renders the dropdown with the overview link 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main main"
  >
    <p
      class="StyledProps--font StyledProps--main text StyledProps--font-xsmall StyledProps--flex StyledProps--flex-alignItems-flex-start StyledProps--flex-justifyContent-space-between"
      style="display: flex; align-items: center;"
    >
      common.accountSetup
      <span
        class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-left-xsmall"
        icon="chevron-down"
      >
        <svg
          data-icon="chevron-down"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            chevron-down
          </desc>
          <path
            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </p>
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
    >
      <a
        class="link"
        href="/account/123/home/setup/"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          overview
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/authentication"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          authentication
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/connectors"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          connectorsLabel
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/secrets"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.secrets
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/access-control"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          accessControl
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/subscriptions"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.subscriptions.title
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/organizations"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          orgsText
        </p>
      </a>
    </div>
  </div>
</div>
`;

exports[`Account Setup Menu it renders the dropdown with the secrets link 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main main"
  >
    <p
      class="StyledProps--font StyledProps--main text StyledProps--font-xsmall StyledProps--flex StyledProps--flex-alignItems-flex-start StyledProps--flex-justifyContent-space-between"
      style="display: flex; align-items: center;"
    >
      common.accountSetup
      <span
        class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-left-xsmall"
        icon="chevron-down"
      >
        <svg
          data-icon="chevron-down"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            chevron-down
          </desc>
          <path
            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </p>
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
    >
      <a
        class="link"
        href="/account/123/home/setup/"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          overview
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/authentication"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          authentication
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/connectors"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          connectorsLabel
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/secrets"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.secrets
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/access-control"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          accessControl
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/subscriptions"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.subscriptions.title
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/organizations"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          orgsText
        </p>
      </a>
    </div>
  </div>
</div>
`;

exports[`Account Setup Menu it renders the dropdown with the subscriptions link 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main main"
  >
    <p
      class="StyledProps--font StyledProps--main text StyledProps--font-xsmall StyledProps--flex StyledProps--flex-alignItems-flex-start StyledProps--flex-justifyContent-space-between"
      style="display: flex; align-items: center;"
    >
      common.accountSetup
      <span
        class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-left-xsmall"
        icon="chevron-down"
      >
        <svg
          data-icon="chevron-down"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            chevron-down
          </desc>
          <path
            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </p>
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
    >
      <a
        class="link"
        href="/account/123/home/setup/"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          overview
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/authentication"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          authentication
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/connectors"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          connectorsLabel
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/secrets"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.secrets
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/access-control"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          accessControl
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/subscriptions"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          common.subscriptions.title
        </p>
      </a>
      <a
        class="link"
        href="/account/123/home/setup/organizations"
      >
        <p
          class="StyledProps--font StyledProps--main text"
        >
          orgsText
        </p>
      </a>
    </div>
  </div>
</div>
`;
